src/edu/csus/ecs/pc2/core/list/UnavailableRunsList.java

```diff
@@ -0,0 +1,202 @@
+        //  of the number of times that run has ever been added to the UnavailableRunsList.
+        int timesAddedToRunList;
+        if (allAddedRuns.containsKey(run.getElementId())) {
+            
+            //the run has previously been added; get the count of number of times it has been added previously
+            timesAddedToRunList = allAddedRuns.get(run.getElementId());
+            
+            //indicate another addition is occurring
+            timesAddedToRunList++ ;
+            allAddedRuns.put(run.getElementId(), timesAddedToRunList);
+            
+        } else {
+            
+            //this is the first time the run has been added
+            allAddedRuns.put(run.getElementId(),  1);
+            timesAddedToRunList = 1;
+        }
+        
+        //compute how long past "now", as a function of number of times already added, the run should "expire" on the UnavailableRunsList.
+        // (Note that the current calculation is a LINEAR increase in time; it would probably be better to have something like an EXPONENTIAL
+        //   increase.  However, since this entire class is (hopefully) temporary, I didn't spend any more time on that... jlc)
+        long contestTimeNowInSecs = contest.getContestTime().getElapsedSecs();
+        long expirationTimeInSecs = contestTimeNowInSecs + (INITIAL_EXPIRATION_TIME_OFFSET_SECS * timesAddedToRunList) ;
+        long expirationLengthInSecs = expirationTimeInSecs - contestTimeNowInSecs;
+        
+        //add the run to the priority queue (note that method add(run) adds the run into the queue in priority order based on the expiration time in the run)
+        UnavailableRun runToAdd = new UnavailableRun(run, expirationTimeInSecs);
+        unavailableRuns.add(runToAdd);
+        
+        if (log!=null) { 
+            log.info("Added run " + run.getNumber() + " to UnavailableRuns list (times added = " + timesAddedToRunList 
+                            + "; time until expiration = " + expirationLengthInSecs + " secs); list size = " + unavailableRuns.size());
+        } else {
+            System.err.println("Added run " + run.getNumber() + " to UnavailableRuns list (times added = " + timesAddedToRunList + ") but no log was available for logging!!");
+            throw new RuntimeException("Added run " + run.getNumber() + " to UnavailableRuns list (times added = " + timesAddedToRunList + ") but no log was available for logging!!");            
+        }
+    }
+    
+    /**
+     * This method returns an indication of whether the specified run is currently contained in the list of "unavailable runs".
+     * 
+     * @param run the run to be looked for in the UnavailableRuns list.
+     * @return true if the run is in the unavailable runs list; false otherwise.
+     */
+    public boolean contains(Run run) {
+        
+        //get an iterator over all the Unavailable Runs, without actually removing any of them from the priority queue        
```

Comment:
Is there a reason to not have a single line for this method?
   return unavailableRuns.containsKey(run.getElementId());
A reason like this is more thread-safe or some potential problem accessing the list using contains() ?

Reply:
You're suggesting changing the method body to instead be
```
return unavailableRuns.containsKey(run.getElementId());
```
?
The "unavailableRuns" object is of type `PriorityBlockingQueue`; it's not a Hashtable (nor any of its derivatives).  There's no such method as `containsKey()` in that data type.

I'm marking this comment as "resolved" because it suggests an invalid change.